### PR TITLE
also fix singleton config for core module

### PR DESCRIPTION
**Pull request by christophsturm**
follow up to #3058 that converts the core module too. 
and unit test it all

### CODE REVIEW

**Review by blabno — APPROVED** [review]
The code looks good.
I see pros and cons in both approaches. Just as forgetting to configure class as singleton in a module you can forget about annotating the class. Defining life cycle inside module allows for class reuse in different context, but that's very theoretical benefit.

File: core/src/main/java/bisq/core/CoreModule.java [hunk 1]
```diff
@@ -118,15 +84,6 @@ protected void configure() {
         String referralId = environment.getProperty(AppOptionKeys.REFERRAL_ID, String.class, "");
         bind(String.class).annotatedWith(Names.named(AppOptionKeys.REFERRAL_ID)).toInstance(referralId);
 
-        bind(MobileNotificationService.class).in(Singleton.class);
-        bind(MobileMessageEncryption.class).in(Singleton.class);
```
Comment (sqrrm) [inline core/src/main/java/bisq/core/CoreModule.java:122]: MobileMessageEncryption doesn't have singleton annotation and not tested in GuiceSetupTest.java

**Review by sqrrm — CHANGES_REQUESTED** [review]
Please fix the missing singleton

**Review by sqrrm — APPROVED** [review]
utACK

**Pull request merged**